broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -249,45 +343,229 @@ class ServiceFabrikApiController extends FabrikBaseController {
           .invoke()
           .tap(response => logger.info('backup response ', response))
           .then(body => res
-            .status(202)
+            .status(CONST.HTTP_STATUS_CODE.ACCEPTED)
             .send(body)
           );
       });
   }
 
+  getBackupOptions(backup_guid, deployment, req) {
+    return Promise
+      .all([
+        cf.cloudController.findServicePlanByInstanceId(req.params.instance_id),
+        cf.cloudController.getOrgAndSpaceGuid(req.params.instance_id)
+      ])
+      .spread((plan_details, res) => {
+        const context = req.body.context || {
+          space_guid: res.space_guid,
+          platform: 'cloudfoundry'
+        };
+        const backupOptions = {
+          guid: backup_guid,
+          deployment: deployment,
+          instance_guid: req.params.instance_id,
+          plan_id: req.body.plan_id || plan_details.entity.unique_id,
+          service_id: req.body.service_id || this.getPlan(plan_details.entity.unique_id).service.id,
+          context: context
+        };
+        return backupOptions;
+      });
+  }
+
+  startBackup_sf20(req, res) {
+    let backup_started_at;
+    let lockedDeployment = false; // Need not unlock if checkQuota fails for parallelly triggered on-demand backup
+    req.manager.verifyFeatureSupport(CONST.OPERATION_TYPE.BACKUP);
+    const trigger = _.get(req.body, 'trigger', CONST.BACKUP.TRIGGER.ON_DEMAND);
+    let backupGuid;
+    return Promise
+      .try(() => this.checkQuota(req, trigger))
+      .then(() => Promise.all([utils
+        .uuidV4(),
+        req.manager
```

Comment:
Deployment should not be passed from here. Instead the backup manager should find it out.